pyreisejl/utility/call.py

```diff
@@ -34,109 +27,16 @@ def _record_scenario(scenario_id, runtime):
-        validate_time_range(start_ts, min_ts, max_ts)
-        validate_time_range(end_ts, min_ts, max_ts)
-
-        if start_ts > end_ts:
-            raise InvalidDateArgument(
-                f"The start date ({start_ts}) cannot be after the end date ({end_ts})."
-            )
-
-        # Julia starts at 1
-        start_index = dates.get_loc(start_ts) + 1
-        end_index = dates.get_loc(end_ts) + 1
-
-        # Calculate number of intervals
-        ts_range = end_index - start_index + 1
-        if ts_range % interval > 0:
-            raise InvalidInterval(
-                "This interval does not evenly divide the given date range."
-            )
-        self.start_index = start_index
-        self.interval = interval
-        self.n_interval = int(ts_range / interval)
-        self.input_dir = input_dir
-        print("Validation complete!")
-
-    def _print_settings(self):
-        print("Launching scenario with parameters:")
-        print(
-            {
-                "interval": self.interval,
-                "n_interval": self.n_interval,
-                "start_index": self.start_index,
-                "input_dir": self.input_dir,
-                "execute_dir": self.execute_dir,
-                "threads": self.threads,
-            }
-        )
-
-    def launch_scenario(self):
-        # This should be defined in sub-classes
-        raise NotImplementedError
-
-
-class GurobiLauncher(Launcher):
-    def launch_scenario(self, execute_dir=None, threads=None, solver_kwargs=None):
-        """Launches the scenario.
-
-        :param None/str execute_dir: directory for execute data. None defaults to an
-            execute folder that will be created in the input directory
-        :param None/int threads: number of threads to use.
-        :param None/dict solver_kwargs: keyword arguments to pass to solver (if any).
-        :return: (*int*) runtime of scenario in seconds
-        """
-        self.execute_dir = execute_dir
-        self.threads = threads
-        self._print_settings()
-        # Import these within function because there is a lengthy compilation step
-        from julia.api import Julia
-
-        Julia(compiled_modules=False)
-        from julia import Gurobi  # noqa: F401
-        from julia import REISE
-
-        start = time()
-        REISE.run_scenario_gurobi(
-            interval=self.interval,
-            n_interval=self.n_interval,
-            start_index=self.start_index,
-            inputfolder=self.input_dir,
-            outputfolder=self.execute_dir,
-            threads=self.threads,
-        )
-        end = time()
-
-        runtime = round(end - start)
-        hours, minutes, seconds = sec2hms(runtime)
-        print(f"Run time: {hours}:{minutes:02d}:{seconds:02d}")
-
-        return runtime
+    launch_map = {"gurobi": GurobiLauncher, "glpk": GLPKLauncher}
+    if solver is None:
+        return GurobiLauncher
+    return launch_map[solver]
```

Comment:
Should we provide a meaningful exception in case the user specifies something besides the keys in `launch_map`?

Should we allow a little more flexibility so that strings like `"Gurobi"` or `"GLPK"` still resolve successfully?

Should we define `launch_map` within the body of **launchers.py** so that we can do a single import?

Reply:
Affirmative, all good ideas. I probably could have made this a draft PR since it's a bit rough currently.

Reply:
I like where it's headed so far.